aoe-web-backend/src/search/es.ts

```diff
@@ -93,14 +96,33 @@ export const createEsIndex = async (): Promise<any> => {
  * @param index
  */
 export const deleteIndex = async (index: string): Promise<boolean> => {
-  return client.indices.delete({
-    index: index
-  }).then((data: any) => {
-    return !!data.body;
-  }).catch((error: any) => {
+  try {
+    const deleteResponse = await client.indices.delete({ index });
+    winstonLogger.info(
+      `Index ${index} deleted with status code: ${deleteResponse.statusCode} and acknowledged: ${deleteResponse.body.acknowledged}`
+    );
+
+    if (!deleteResponse.body.acknowledged) {
+      return false;
+    }
+
+    for (let attempt = 0; attempt < 5; attempt++) {
+      const exists = await indexExists(index);
+      if (!exists) {
+        winstonLogger.info(`Confirmed that index ${index} no longer exists.`);
+        return true;
+      }
+      winstonLogger.warn(`Index ${index} still exists after attempt ${attempt}.`);
+
+      await new Promise((resolve) => setTimeout(resolve, 2000));
```

Comment:
mietin että olisiko tässä hyvä laittaa pientä exponential backoffia, tyyliin

```await new Promise((resolve) => setTimeout(resolve, (attempt + 1) * 2000))```

tässä tuo attempt voi olla ekalla kerralla 0, koska for-loop tekee post-incrementin, siksi tuo `+ 1`.